utils/git_diff.py

```diff
@@ -0,0 +1,326 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import os
+import re
+from contextlib import contextmanager
+from pathlib import Path
+
+from git import Repo
+
+
+@contextmanager
+def checkout_commit(repo, commit_id):
+    """
+    Context manager that checks out a commit in the repo.
+    """
+    current_head = repo.head.commit if repo.head.is_detached else repo.head.ref
+
+    try:
+        repo.git.checkout(commit_id)
+        yield
+
+    finally:
+        repo.git.checkout(current_head)
+
+
+def diff_is_docstring_only(repo, branching_point, filename):
+    """
+    Check if the diff is only in docstrings in a filename.
+    """
+    with checkout_commit(repo, branching_point):
+        with open(filename, "r", encoding="utf-8") as f:
+            old_content = f.read()
+
+    with open(filename, "r", encoding="utf-8") as f:
+        new_content = f.read()
+
+    # To check if the diff is in docstrings, we remove all docstrings by splitting on """.
```

Comment:
That's smart! Nitpick, no need to update, it could also run on `'''` and see if the line starts with `#`!